HZETRN/btrn.py

```diff
@@ -0,0 +1,377 @@
+start_vol  The volume containing the ref_point
+ref_point  the origin of the ray
+dir 	   the direction of the ray
+fname_for_vol a mapping of vol id to material object
+
+returns    arrays  for length and material name as the ray
+           travels from the start point to the graveyard
+
+This depends on dagmc's ray_iterator, which calls ray_fire
+"""
+def slabs_for_ray(start_vol, ref_point, dir, fname_for_vol):
+    is_graveyard = False
+    foundMat = False
+    surf = 0
+    dist = 0
+    huge = 1000000000
+
+    slab_lengths = []
+    slab_mat_names = []
+    vols_traversed = []
+
+    # Could be starting in the implicit_complement
+    if start_vol in fname_for_vol:
+        #  Done before we've started
+        if fname_for_vol[start_vol] == 'graveyard':
+	    # print 'Starting volume is graveyard'
+            return slab_length, slab_mat_name 
+    else:
+        print 'We are starting in the implicit complement'
+    is_graveyard = False    
+    last_vol = start_vol
+    
+    for (next_vol, dist, surf, xyz) in dagmc.ray_iterator(last_vol, ref_point, dir, yield_xyz=True):
+        # print 'next_vol', next_vol
+        if not is_graveyard and ((dist < huge) and (surf != 0)):
+	    # We have already checked last_vol for being a graveyard
+	    if last_vol in fname_for_vol:
+	        slab_lengths.append(dist)
+		# Get a name associated with volume
+	        slab_mat_names.append(fname_for_vol[last_vol])
+		# Optional documentation
+                vols_traversed.append(last_vol)
+
+	    if next_vol in fname_for_vol:
+   	        if fname_for_vol[next_vol] == 'graveyard':
+	            is_graveyard = True
+
+            # If last_vol is the implicit compl., this is the only line executed
+            last_vol = next_vol
+
+	# We hit the graveyard
+        else:
+	    break
+
+    #########################################################################
+    # sdir = [format(dir[0],'.6f'),format(dir[1], '.6f'), format(dir[2], '.6f')]
+    #########################################################################
+
+    return slab_lengths, slab_mat_names
+      
+"""
+Argument parsing
+returns : args: -f for the input geometry file, -r for the input ray tuple file
+ref     : DAGMC/tools/parse_materials/dagmc_get_materials.py
+"""
+def parsing():
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument(
+        '-f', action='store', dest='uwuw_file', 
+	help='The relative path to the .h5m geometry file')
+
+    parser.add_argument(
+        '-d', action='store', dest='run_dir',
+	help='The name of the holding directory for all hzetrn runs')
+
+    parser.add_argument(
+        '-e', action='store', dest='rad_env', 
+	help='The radiation environment: spe (default) or gcr')
+    parser.set_defaults(rad_env='spe')
+
+    parser.add_argument(
+        '-r', action='store', dest='ray_dir_file', 
+	help='The path to the file with ray direction tuples')
+    parser.set_defaults(ray_dir_file='')
+
```

Comment:
may want to change help statement for -r and -n, if -r not provided then fire -n random rays, if both fire n ray subset from file